ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -565,49 +577,335 @@ toPublicState PeerSelectionState { knownPeers
+      numberOfEstablishedLocalRootPeers          = Set.size $ establishedLocalRootsPeersSet,
+      numberOfWarmLocalRootPeersPromotions       = Set.size $ establishedLocalRootsPeersSet `Set.intersection` inProgressPromoteWarm,
+      numberOfActiveLocalRootPeers               = Set.size   activeLocalRootsPeersSet,
+      numberOfActiveLocalRootPeersDemotions      = Set.size $ activeLocalRootsPeersSet `Set.intersection` inProgressDemoteHot,
+
+      localRoots,
+
+      numberOfKnownSharedPeers                   = Set.size   knownSharedPeersSet,
+      numberOfColdSharedPeersPromotions          = Set.size $ knownSharedPeersSet `Set.intersection` inProgressPromoteCold,
+      numberOfEstablishedSharedPeers             = Set.size   establishedSharedPeersSet,
+      numberOfWarmSharedPeersDemotions           = Set.size $ establishedSharedPeersSet `Set.intersection` inProgressDemoteWarm,
+      numberOfWarmSharedPeersPromotions          = Set.size $ establishedSharedPeersSet `Set.intersection` inProgressPromoteWarm,
+      numberOfActiveSharedPeers                  = Set.size   activeSharedPeersSet,
+      numberOfActiveSharedPeersDemotions         = Set.size $ activeSharedPeersSet `Set.intersection` inProgressDemoteHot
     }
   where
-    knownPeersSet = KnownPeers.toSet (knownPeers st)
-    establishedPeersSet = EstablishedPeers.toSet (establishedPeers st)
-    coldPeersSet  = knownPeersSet Set.\\ establishedPeersSet
-    warmPeersSet  = establishedPeersSet Set.\\ activePeers
-    hotPeersSet   = activePeers
+    -- convention: only `{knwon,established,active}.*PeersSet` and
+    -- `inProgress{Promote,Demote}{Cold,Warm,Hot}` identifiers are used in the
+    -- calculations above ↑
+
+    -- common sets
+    knownSet       = KnownPeers.toSet knownPeers
+    establishedSet = EstablishedPeers.toSet establishedPeers
+    bigLedgerSet   = PublicRootPeers.getBigLedgerPeers publicRootPeers
+
+    -- non big ledger peers
+    knownPeersSet       = knownSet Set.\\ bigLedgerSet
+    establishedPeersSet = establishedSet Set.\\ bigLedgerSet
+    activePeersSet      = activePeers Set.\\ bigLedgerSet
+
+    -- big ledger peers
+    knownBigLedgerPeersSet       = knownSet `Set.intersection` bigLedgerSet
+    establishedBigLedgerPeersSet = establishedPeersSet `Set.intersection` bigLedgerSet
+    activeBigLedgerPeersSet      = activePeers `Set.intersection` bigLedgerSet
+
+    -- local roots
     localRoots =
       [ (hot, warm)
       | (hot, warm, _) <- LocalRootPeers.toGroupSets localRootPeers
       ]
-    bigLedgerPeers = PublicRootPeers.getBigLedgerPeers publicRootPeers
+    localRootSet                  = LocalRootPeers.keysSet localRootPeers
+    knownLocalRootPeersSet        = knownPeersSet `Set.intersection` localRootSet
+    establishedLocalRootsPeersSet = establishedPeersSet `Set.intersection` localRootSet
+    activeLocalRootsPeersSet      = activePeers `Set.intersection` localRootSet
```

Comment:
I think this doesn't matter, because a local root peers are disjoint from big ledger peers: [ref](https://github.com/intersectmbo/ouroboros-network/blob/77cb6fee1ed7788e7202df17b2273226d0e86388/ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs#L1030).

Reply:
I'll follow your suggestion, because `activePeersSet` is smaller.